lib/utils.pm

```diff
@@ -2232,4 +2233,32 @@ sub package_upgrade_check {
     }
 }
 
+=head2 gcloud_install
+    gcloud_install();
+
+This function is used to install the gcloud CLI 
+for the GKE Google Cloud.
+
+From $url we get the full package and install in $dir local folder,
+that is supposed to be a subdir of the root user home /root.
+The Defaults are availeble for a simple call without parameters.
+
+=cut
+
+sub gcloud_install {
+    my %args = @_;
+    my $url = $args{url} || 'sdk.cloud.google.com';
+    my $dir = $args{dir} || 'google-cloud-sdk';
+    my $timeout = $args{timeout} || 700;
+
+    assert_script_run("zypper -n in curl tar gzip", $timeout);
+    assert_script_run("export CLOUDSDK_CORE_DISABLE_PROMPTS=1");
+    assert_script_run("curl $url | bash", $timeout);
+
+    assert_script_run("echo . /root/$dir/completion.bash.inc >> ~/.bashrc");
+    assert_script_run("echo . /root/$dir/path.bash.inc >> ~/.bashrc");
+    record_info('GCE', script_output('source ~/.bashrc && gcloud version'));
+}
```

Comment:
We also need to reload the `.bashrc` otherwise the newly added include files are not loaded

```suggestion
    assert_script_run('. ~/.bash_profile');
}
```

Reply:
> We also need to reload the `.bashrc` otherwise the newly added include files are not loaded

`.bashrc` is already loaded in line 2260. Consider that a `.bash_profile` is executed only at _login_ of a shell

Reply:
IMHO this will not work - Look at http://marvin5.arch.suse.cz/tests/440#step/helm_GKE/46 - if you run `script_output` or `script_run` it is wrapped in some kind of helper script that is executed. This means, that the `source ~./bashrc && gcloud version` will be executed in a new bash, so it has no effect on the current bash.

Only `assert_script_run` runs the commands on the bare bash, `script_output` does some internal magic so you will not have the intended effect.

***

This is very cryptic to understand, but look at http://marvin5.arch.suse.cz/tests/440#step/helm_GKE/46:

    "echo GR72W; bash -oe pipefail /tmp/scriptGR72W.sh ; echo SCRIPT_FINISHEDGR72W-\$?-"

Your command `source ~/.bashrc && gcloud version` is wrapped into `/tmp/scriptGR72W.sh`, so it will be only present in that script, not in the actual bash running here.

Reply:
done